jwql/website/apps/jwql/views.py

```diff
@@ -271,36 +271,61 @@ def archived_proposals_ajax(request, inst):
     inst = JWST_INSTRUMENT_NAMES_MIXEDCASE[inst.lower()]
     filesystem = get_config()['filesystem']
 
+    # Dictionary to hold summary information for all proposals
+    all_proposals = get_instrument_proposals(inst)
+    all_proposal_info = {'num_proposals': 0,
+                         'proposals': [],
+                         'thumbnail_paths': [],
+                         'num_files': []}
+
     # Get list of all files for the given instrument
-    filename_query = mast_query_filenames_by_instrument(inst)
-    filenames_public = get_filenames_by_instrument(inst, restriction='public', query_response=filename_query)
-    filenames_proprietary = get_filenames_by_instrument(inst, restriction='proprietary', query_response=filename_query)
-
-    # Determine locations to the files
-    filenames = []
-    for filename in filenames_public:
-        try:
-            relative_filepath = filesystem_path(filename, check_existence=False)
-            full_filepath = os.path.join(filesystem, 'public', relative_filepath)
-            filenames.append(full_filepath)
-        except ValueError:
-            print('Unable to determine filepath for {}'.format(filename))
-    for filename in filenames_proprietary:
-        try:
-            relative_filepath = filesystem_path(filename, check_existence=False)
-            full_filepath = os.path.join(filesystem, 'proprietary', relative_filepath)
-            filenames.append(full_filepath)
-        except ValueError:
-            print('Unable to determine filepath for {}'.format(filename))
-
-    # Gather information about the proposals for the given instrument
-    proposal_info = get_proposal_info(filenames)
+    for proposal in all_proposals:
```

Comment:
This is just where we are breaking the views into proposals based on instrument and propid? This looks more efficient 👍

Reply:
Yes, I like this much better. This pushes more of the filtering work onto MAST, which is faster than when we do it ourselves.